### ISSUE TITLE
Route::resource not translate url

### ISSUE THREAD

**Issue by ppalella** [ARCANEDEV/Localization#14]
I need to translate my entirely url and try with `Route::resource('routes.users', 'UsersController');` but return 404 page :(

My routes.php file:

```
Route::localizedGroup(function () {
    Route::transGet('routes.welcome', function () { return view('welcome'); });
    Route::resource('routes.users', 'UsersController');
});
```

Results:
en/welcome **work**
es/bienvenido **work**
en/users **fail**
es/usuarios **fail**

Try change line `Route::resource('routes.users', 'UsersController');` to `Route::resource('users', 'UsersController');`
en/users **work**
es/usuarios **fail**

**Comment by arcanedev-maroc** [comment]
The `Route::resource()` is not in **Localization** package scope.

As you can see there is no `Route::transResource`, and the `routes.users` is not going to be translated.

Try to be explicit with your routes and use the `transGet`, `transPost`, `transDelete` ... methods, that will fix your problem.

Check : https://github.com/ARCANEDEV/Localization/issues/9

**Comment by ppalella** [comment]
Do you plan to add this feature (`Route::transResource`) to the next commit?

**Comment by arcanedev-maroc** [comment]
I don't know :confused: , and to be honest with you, it's not a good idea.

The `Route::resource` may unintentionally take precedence over your supplemental routes.

And `Route::resource` + The **Localization** package = Nightmare.

_If you want my advise_, try to be explicit with your routes. It's like you're drawing paths (roads/routes) to your controllers.

**Comment by ppalella** [comment]
Ok, I understand... I will change my code. Thanks a lot!

**Comment by arcanedev-maroc** [comment]
You're welcome :+1:

**Comment by f-liva** [comment]
I think that **transResource** should be implemented, as Laravel provides **Route::resource** your package should support it, or at least provide a method like **LaravelLocalization::transRoute** that allows you to translate the route on the fly as on mcamara and make the resource work.

By not supporting **Route::transResource** you are discouraging the use of Laravel's good practices, which suggests using Route::resource in case of certain needs.

How can I help?

**Comment by arcanedev-maroc** [comment]
Hi @fede91it,

**TLDR: Feel free to make a PR with tests** :+1: 

TBH, i'm not fan about the `Route::resource` because i prefer to create routes explicitly.

This is why the `transResource` method is not available.

> By not supporting Route::transResource you are discouraging the use of Laravel's good practices, which suggests using Route::resource in case of certain needs.

I don't think this considered as a **Laravel's good practices**, this is just what we call a **magic functionnality** that allows you to do a lot of things & faster with little efforts.

The idea behind `Route::resource` method is to create the **RESTful** routes. 
*You can achieve that by  using explicit routes*

Wiki: https://en.wikipedia.org/wiki/Representational_state_transfer

My top 4 benefits about using explicit routes are:

 * 100% Total control of what i'm doing.
 * Easy route customization (without cheking the laravel's documentations or forums).
 * By reading the routes file(s), i have a clear view of the available routes without using the `php artisan route:list` command.
 * Prevents me to be a lazy person :stuck_out_tongue_closed_eyes: 

This is an example how i use & structure the routes in a OOP way in one of my projects:
   * [RouteServiceProvider](https://github.com/ARCANESOFT/Auth/blob/master/src/Providers/RouteServiceProvider.php#L59)
   * [UsersRoutes](https://github.com/ARCANESOFT/Auth/blob/master/src/Http/Routes/Admin/UsersRoutes.php)

This is just my opinion and i'm not against your suggestion, so feel free to make a PR to add this feature.
